File: src-tauri/src/cmd/webdav.rs
use crate::models::{
    ChangeLogEntry, SecureAccount, SyncConflict, SyncResult, WebdavConfig, WebdavSyncStatus,
};
use crate::webdav;
use tauri::AppHandle;

#[tauri::command]
pub fn webdav_save_config(app: AppHandle, config: WebdavConfig) -> Result<(), String> {
    webdav::save_webdav_config(&app, &config)
}

#[tauri::command]
pub fn webdav_load_config(app: AppHandle) -> Result<Option<WebdavConfig>, String> {
    Ok(webdav::load_webdav_config(&app))
}

#[tauri::command]
pub fn webdav_test_connection(config: WebdavConfig) -> Result<(), String> {
    webdav::test_webdav_connection(&config)
}

#[tauri::command]
pub fn webdav_get_status(app: AppHandle) -> Result<WebdavSyncStatus, String> {
    webdav::get_sync_status(&app)
}

#[tauri::command]
pub fn webdav_start_sync(app: AppHandle) -> Result<SyncResult, String> {
    webdav::sync_now(&app)
}

#[tauri::command]
pub fn webdav_sync_entry(app: AppHandle, entry: SecureAccount) -> Result<(), String> {
    webdav::sync_entry_to_remote(&app, &entry)
}

#[tauri::command]
pub fn webdav_pull_changes(app: AppHandle) -> Result<Vec<SyncConflict>, String> {
    webdav::pull_changes(&app)
}

#[tauri::command]
pub fn webdav_get_changelog(app: AppHandle) -> Result<Vec<ChangeLogEntry>, String> {
    Ok(webdav::get_changelog(&app))
}

#[tauri::command]
pub fn webdav_start_polling(app: AppHandle) -> Result<(), String> {
    webdav::start_polling(app);
    Ok(())
}

#[tauri::command]
pub fn webdav_stop_polling() -> Result<(), String> {
    webdav::stop_polling();
    Ok(())
}

#[tauri::command]
pub fn webdav_stop_sync() -> Result<(), String> {
    webdav::stop_polling();
    Ok(())
}

File: tools/codebook/src/api/webdav.ts
import { invoke } from '@tauri-apps/api/core'
import type { Account } from '../types/codebook'
import type { WebdavConfig, WebdavSyncStatus, SyncConflict, ChangeLogEntry, SyncResult } from '../types/webdav'

export const saveWebdavConfig = async (config: WebdavConfig) => {
  return invoke<void>('webdav_save_config', { config })
}

export const loadWebdavConfig = async () => {
  return invoke<WebdavConfig | null>('webdav_load_config')
}

export const testWebdavConnection = async (config: WebdavConfig) => {
  return invoke<void>('webdav_test_connection', { config })
}

export const getWebdavStatus = async () => {
  return invoke<WebdavSyncStatus>('webdav_get_status')
}

export const startWebdavSync = async () => {
  return invoke<SyncResult>('webdav_start_sync')
}

export const syncEntryToRemote = async (entry: Account) => {
  return invoke<void>('webdav_sync_entry', { entry })
}

export const pullWebdavChanges = async () => {
  return invoke<SyncConflict[]>('webdav_pull_changes')
}

export const getWebdavChangelog = async () => {
  return invoke<ChangeLogEntry[]>('webdav_get_changelog')
}

export const startWebdavPolling = async () => {
  return invoke<void>('webdav_start_polling')
}

export const stopWebdavPolling = async () => {
  return invoke<void>('webdav_stop_polling')
}
